rust/chains/hyperlane-ethereum/src/fallback.rs

```diff
@@ -97,30 +100,36 @@ impl JsonRpcClient for FallbackProvider<PrometheusJsonRpcClient<Http>> {
         let params = serde_json::to_value(params).expect("valid");
 
         let mut errors = vec![];
-        for (idx, provider) in self.0.iter().enumerate() {
-            let fut = match params {
-                Value::Null => provider.request(method, ()),
-                _ => provider.request(method, &params),
-            };
-
-            match fut.await {
-                Ok(v) => return Ok(serde_json::from_value(v)?),
-
-                Err(HttpClientError::ReqwestError(e)) => {
-                    warn!(error=%e, provider_index=%idx, ?provider, method, "ReqwestError in http provider; falling back to the next provider");
-                    errors.push(HttpClientError::ReqwestError(e).into())
-                }
-                Err(HttpClientError::SerdeJson { err, text }) => {
-                    warn!(error=%err, text, provider_index=%idx, ?provider, method, "ReqwestError in http provider; falling back to the next provider");
-                    errors.push(HttpClientError::SerdeJson { err, text }.into())
-                }
-                Err(HttpClientError::JsonRpcError(e)) => {
-                    if METHODS_TO_NOT_TO_FALLBACK_ON.contains(&method) {
-                        warn!(error = %e, provider_index=%idx, ?provider, method, "JsonRpcError in http provider; not falling back");
-                        return Err(HttpClientError::JsonRpcError(e).into());
-                    } else {
-                        warn!(error = %e, provider_index=%idx, ?provider, method, "JsonRpcError in http provider; falling back to the next provider");
-                        errors.push(HttpClientError::JsonRpcError(e).into())
+        // make sure we do at least 4 total retries.
+        while errors.len() <= 3 {
+            if !errors.is_empty() {
+                sleep(Duration::from_secs(10)).await;
```

Comment:
Hmm, good point. I had done this because with the fallback provider I was running into constant crashes because one of the chins we only specified one provider for kept returning errors and I just wanted to put some gap inbetween. Another option would be to allow configuring what the duration is at construction. I will at least decrease if not make it configurable.